src/net/tcp/listener.rs

```diff
@@ -53,6 +57,25 @@ impl TcpListener {
         Ok(TcpListener { inner: socket })
     }
 
+    /// Creates new `TcpListener` from a previously bound `std::net::TcpListener`.
+    ///
+    /// This function is intended to be used to wrap a TCP listener from the
+    /// standard library in the tokio-uring equivalent. The conversion assumes nothing
+    /// about the underlying socket; it is left up to the user to decide what socket
+    /// options are appropriate for their use case.
+    ///
+    /// This can be used in conjunction with socket2's `Socket` interface to
+    /// configure a socket before it's handed off, such as setting options like
+    /// `reuse_address` or binding to multiple addresses.
+    pub fn from_std(socket: std::net::TcpListener) -> Self {
```

Comment:
Can you explain to me why it's a bad idea to allow this in all cases?

Reply:
I'm not sure actually. I've been considering doing away with all types derived from FD myself anyway in another repo as I figure the kernel is dealing with integers and has to protect itself and the types we create on top of it just seem artificial at times.

I guess I've seen so many requests for help with code that didn't work when the cause could be attributed to using `unsafe` incorrectly, and not documenting it, that if there were another way to get the user's to document their understanding of why it was logical to create a higher order type from a lower ordered one, it could only help.

But if you feel that's wrong, I'm okay with leaving it defined the way you did. I wasn't sure myself.

Anyway, anything that modifies the types I'll defer to @Noah-Kennedy anyway in this repo.

Reply:
I agree with you. But since this is the same type, but from another library, it should be fine to transfer ownership of the FD, which is what `UdpSocket` does. So this should be fool-proof But then again, marking it unsafe would warn the user, as with `FromRawFd`.